python/rmm/_lib/memory_resource.pyx

```diff
@@ -297,13 +297,14 @@ cpdef void set_current_device(int device) except *:
         raise RuntimeError(f"Failed to set CUDA device with error: {err}")
 
 
-cpdef _initialize(
+cpdef void _initialize(
     bool pool_allocator=False,
     bool managed_memory=False,
     object initial_pool_size=None,
     object devices=0,
     bool logging=False,
     object log_file_name=None,
+    bool cuda_initialization=True,
 ):
```

Comment:
If this returns `void`, shouldn't we also note Python exceptions may occur? Also are C++ exceptions a concern?

Reply:
Marked as `except *` since there's no direct usage of externed functions and the wrappers we use are already marked as `except +` where appropriate.

Reply:
Sounds good. Wasn't sure about C++ exceptions. So had included both just in case.